backend/fms_core/services/samplesheet.py

```diff
@@ -0,0 +1,78 @@
+from django.templatetags.static import static
+from django.conf import settings
+
+import os
+from io import BytesIO
+from collections import defaultdict
+from openpyxl.reader.excel import load_workbook
+
+from fms_core.coordinates import convert_alpha_digit_coord_to_ordinal
+from fms_core.containers import CONTAINER_KIND_SPECS
+from fms_core.models import DerivedBySample, Index
+
+SAMPLESHEET_FILE_PATH = static("/samplesheet_templates/SampleSheet_v2.xlsx")
+
+def get_samplesheet(container_kind, placement):
+    """
+    Prefill a samplesheet V2 using samples index information and return it as a Byte stream.
+
+    Args:
+        `container_kind`: The reported container kind sent with the request. Used to calculate lanes.
+        `placement`: List holding each lane mapping between the coordinate and the pooled sample occupying the coordinate.
+                    exemple: [{"coordinates": "A01", "sample": "1041243"}, {"coordinates": "B01", "sample": "1414432"}, ...]
+    
+    Returns:
+        `samplesheet`: Samplesheet file containing the prefiled samples and index in excel format.
+        `errors`     : Errors raised during creation of the samplesheet file.
+        `warnings`   : Warnings raised during creation of the samplesheet file.
+
+    """
+    out_stream = BytesIO()
+    errors = []
+    warnings = []
+
+    SAMPLESHEET_SHEET_NAME = "Samplesheet"
+    SAMPLESHEET_FIRST_INPUT_ROW = 23
+
+    COLUMN_LANE = 1
+    COLUMN_SAMPLE_ALIAS = 2
+    COLUMN_INDEX_3PRIME = 3
+    COLUMN_INDEX_5PRIME = 4
+
+    row_data_by_lane = defaultdict(list)
+    try:
+        filename = "/".join(SAMPLESHEET_FILE_PATH.split("/")[-2:]) # Remove the /static/ from the served path to search for local path 
+        template_path = os.path.join(settings.STATIC_ROOT, filename)
+        workbook = load_workbook(filename=template_path)
+        sheet = workbook[SAMPLESHEET_SHEET_NAME]
+        container_spec = CONTAINER_KIND_SPECS[container_kind]
+        if container_spec is None:
+            raise Exception(f'Cannot convert coord to lane number. No ContainerSpec found for container kind "{container_kind}".')
+        for lane_info in placement:
```

Comment:
What a strange variable name. There's just `lane_info`s inside `placement`?

Reply:
lane info is a dictionary...

Reply:
I'm aware. It's just an odd name.

`placement` is supposedly an array of
```ts
        type PlacementData = {
            coordinates: string,
            sample_id: FMSId,
        }[]
```

I don't see how that has anything to do with lane.

Reply:
placement send data back along the request ... each coordinate on the container has the lane information required for the samplesheet ...